framework/TSA/AutoCorrelation.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Auto Correlation Transformation for Characterization of Time-Series Data.
+"""
+import numpy as np
+
+from utils import InputData, InputTypes, xmlUtils
+from .TimeSeriesAnalyzer import TimeSeriesGenerator, TimeSeriesCharacterizer
+
+
+# utility methods
+class AutoCorrelation(TimeSeriesGenerator , TimeSeriesCharacterizer):
+  """
+    Perform AutoCorrelation Transformation on time-dependent data.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class cls.
+      @ Out, specs, InputData.ParameterInput, class to use for specifying input of cls.
+    """
+    specs = super(AutoCorrelation, cls).getInputSpecification()
+    specs.name = 'AutoCorrelation'
+    specs.description = """AutoCorrelation TimeSeriesAnalysis algorithm."""
```

Comment:
This should probably be expanded to explain how it does what it does and why one might use it, a la user manual entry. You can copy the contents from the UserManual directly if you want.

Reply:
This will be changed to have a better description